pkgs/misc/uboot/default.nix

```diff
@@ -179,27 +181,21 @@ in {
     filesToInstall = ["u-boot-dtb.img" "SPL"];
   };
 
-  # Flashing instructions:
-  # dd if=bl1.bin.hardkernel of=<device> conv=fsync bs=1 count=442
-  # dd if=bl1.bin.hardkernel of=<device> conv=fsync bs=512 skip=1 seek=1
-  # dd if=u-boot.gxbb of=<device> conv=fsync bs=512 seek=97
-  ubootOdroidC2 = let
-    firmwareBlobs = fetchFromGitHub {
-      owner = "armbian";
-      repo = "odroidc2-blobs";
-      rev = "47c5aac4bcac6f067cebe76e41fb9924d45b429c";
-      sha256 = "1ns0a130yxnxysia8c3q2fgyjp9k0nkr689dxk88qh2vnibgchnp";
-      meta.license = lib.licenses.unfreeRedistributableFirmware;
-    };
-  in buildUBoot {
+  ubootOdroidC2 = buildUBoot {
     defconfig = "odroid-c2_defconfig";
-    extraMeta.platforms = ["aarch64-linux"];
-    filesToInstall = ["u-boot.bin" "u-boot.gxbb" "${firmwareBlobs}/bl1.bin.hardkernel"];
+
+    # hack: make board able to boot from eMMC
+    #     : wip https://groups.io/g/u-boot-amlogic/topic/meson_gxbb_regression_since/77816805
+    preBuild = ''
+      sed -i "388d" arch/arm/dts/meson-gxbb.dtsi
```

Comment:
Can we verify the line content before deleting it, for whenever a future u-boot update changes the device tree sources, it'd be best to break in a controlled manner rather than delete an arbitrary line.

Reply:
Yeah, very true. I switched this to *.patch with the relevant comments.